From 77fa6f96abf737b3299a1e3eae9756955195dcc0 Mon Sep 17 00:00:00 2001
From: Yumi Yukimura <me.cafebabe@gmail.com>
Date: Sun, 16 Aug 2026 21:58:02 +0800
Subject: [PATCH 3/3] legacy-um: Fix platform apps label location for ims and
 QtiTelephonyService

ERROR: Platform apps cannot be labeled within vendor's seapp_contexts.
Please fix the following violations by doing one of these.
1) Move these apps to vendor or odm partition, and then label the app with types without 'coredomain'.
2) Move these seapp_contexts entries to system, system_ext, or product partition's seapp_contexts, and then label the apps with 'coredomain' types.
Violations:
  APK name: ims.apk
  package name: org.codeaurora.ims
  seapp_contexts partition: vendor
  line: 'user=_app seinfo=platform name=org.codeaurora.ims isPrivApp=true domain=vendor_qtelephony type=app_data_file levelFrom=all'
 ---
  APK name: QtiTelephonyService.apk
  package name: com.qualcomm.qti.telephonyservice
  seapp_contexts partition: vendor
  line: 'user=_app seinfo=platform name=com.qualcomm.qti.telephonyservice domain=vendor_qtelephony type=app_data_file levelFrom=all'
 ---

******************************
ERROR: SELinux Treble Labeling violations found.
Treble Labeling will be enforced for devices with BOARD_API_LEVEL >= 202604.
Please fix all violations. For more details, please refer to:
system/sepolicy/tests/treble_labeling_tests.md
******************************

Change-Id: I0fc137ce8eebb0e5e174b4b71db4d615f6aeea8d
---
 generic/private/seapp_contexts      | 3 +++
 legacy/vendor/common/seapp_contexts | 6 ------
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/generic/private/seapp_contexts b/generic/private/seapp_contexts
index 85f6e155..165a9aeb 100644
--- a/generic/private/seapp_contexts
+++ b/generic/private/seapp_contexts
@@ -44,3 +44,6 @@ user=_app seinfo=platform name=com.qualcomm.qti.devicestatisticsservice domain=v
 
 # Add new domain for qti alarm app
 user=_app seinfo=platform name=com.qualcomm.qti.poweroffalarm domain=vendor_qti_poweroffalarm_app isPrivApp=false type=app_data_file levelFrom=all
+
+# QtiTelephonyService app
+user=_app seinfo=platform name=com.qualcomm.qti.telephonyservice domain=vendor_qtelephony type=app_data_file levelFrom=all
diff --git a/legacy/vendor/common/seapp_contexts b/legacy/vendor/common/seapp_contexts
index e2dface3..6bd5be9e 100644
--- a/legacy/vendor/common/seapp_contexts
+++ b/legacy/vendor/common/seapp_contexts
@@ -51,15 +51,9 @@ user=system seinfo=platform name=com.qualcomm.qti.perfdump domain=perfdump_app t
 #add new domain for qtidataservices
 user=_app seinfo=platform name=.qtidataservices domain=qtidataservices_app type=app_data_file levelFrom=all
 
-#Add new domain for ims app
-user=_app seinfo=platform name=org.codeaurora.ims isPrivApp=true domain=vendor_qtelephony type=app_data_file levelFrom=all
-
 #Add DeviceInfoHidlClient to vendor_qtelephony
 user=_app seinfo=platform name=com.qualcomm.qti.devicestatisticsservice domain=vendor_qtelephony type=app_data_file levelFrom=all
 
-# QtiTelephonyService app
-user=_app seinfo=platform name=com.qualcomm.qti.telephonyservice domain=vendor_qtelephony type=app_data_file levelFrom=all
-
 #Add ExtTelephonyService to vendor_qtelephony
 user=_app seinfo=platform name=com.qti.phone domain=vendor_qtelephony type=app_data_file levelFrom=all
 
-- 
2.47.3

